src/dsp_tools/commands/xmlupload/models/ontology_diagnose_models.py

```diff
@@ -58,7 +56,10 @@ def execute_problem_protocol(self) -> tuple[str, pd.DataFrame | None]:
         prop_msg = self._compose_problem_string_for_props()
         if prop_msg:
             msg += prop_msg
-        if len(self.classes) + len(self.properties) > maximum_prints:
+        if (
+            self._calculate_num_resources(self.classes) + self._calculate_num_resources(self.properties)
+            > maximum_prints
+        ):
```

Comment:
This could fit in 2 lines instead of 4:
```
problem_count = self._calculate_num_resources(self.classes) + self._calculate_num_resources(self.properties)
        if problem_count > maximum_prints:
```